### PR TITLE
Lots of lang-y CustomModelData-y stuff

### LINKED ISSUE #354: Crossbow Cartridges uses the old format
e.g. lang  and `@p` instead of `@a[limit=1]`

### PR DESCRIPTION
**Pull request by AndanteDevs**
I've only been thinking about doing this for a few months

- converted as much as possible from text to lang
    - `TRANSLATION.KEY.FOR.MISODE`; hah, for Dan more like
- fixed loads of advancements
- removed all those `["",` things
    - what... even... are they?...
- `CustomModelData`
    - because texture people (♥ love u Qbert)

*ignore the name of the branch btw*
Fixes #354

### CODE REVIEW

**Comment by AndanteDevs** [comment]
Also, can this be deleted: [gm4_mysterious_midnights/data/gm4/advancements/mysterious_midnights.json](https://github.com/Gamemode4Dev/GM4_Datapacks/blob/master/gm4_mysterious_midnights/data/gm4/advancements/mysterious_midnights.json)

**Comment by grian32** [comment]
> because texture people (♥ love u Qbert)
i feel offended

File: gm4_ambrosia/_pack.mcmeta [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "pack":{
     "pack_format":0,
-    "description":"A Gamemode 4 Module"
+    "description":{"translate":"pack.gm4.description"}
```
[the remark this answers is not in the document]
Comment (AndanteDevs) [reply gm4_ambrosia/_pack.mcmeta]: 1. it's the same (currently) for all of the modules, so there's no reason not to
2. completely forgot to do the translate-y optional thing, oops

[the remark this answers is not in the document]
Comment (misode) [reply gm4_ambrosia/_pack.mcmeta]: Only admins can see that, so imo it shouldn't be translated

File: gm4_ambrosia/data/gm4_ambrosia/functions/honey_bottle/give_new_bottle/level_1.mcfunction [hunk 2]
```diff
@@ -3,8 +3,8 @@
...
+replaceitem entity @s[gamemode=!creative,tag=gm4_drank_honey_from_mainhand] weapon.mainhand potion{gm4_ambrosia:{item:"honey_bottle",honey_level:1},display:{Name:'{"translate":"%1$s","with":["Almost Empty Bottle of Honey","item.gm4.honey_bottle.1"],"italic":"false"}',Lore:['["1",{"translate":"%1$s%3427655$s","with":["/9 Spoons",{"translate":"item.gm4.honey_bottle.lore.spoons"}]}]']},CustomPotionColor:16754944,CustomPotionEffects:[{Id:10,Amplifier:0,Duration:1600}]}
+replaceitem entity @s[gamemode=!creative,tag=!gm4_drank_honey_from_mainhand,nbt={Inventory:[{Slot:-106b,id:"minecraft:glass_bottle",Count:1b}]}] weapon.offhand potion{gm4_ambrosia:{item:"honey_bottle",honey_level:1},display:{Name:'{"translate":"%1$s","with":["Almost Empty Bottle of Honey",[{"translate":"item.gm4.honey_bottle.level.almost_empty"},{"translate":"item.gm4.honey_bottle"}]],"italic":"false"}',Lore:['["1",{"translate":"%1$s%3427655$s","with":["/9 Spoons",{"translate":"item.gm4.honey_bottle.lore.spoons"}]}]']},CustomPotionColor:16754944,CustomPotionEffects:[{Id:10,Amplifier:0,Duration:1600}]}
```
[the remark this answers is not in the document]
Comment (AndanteDevs) [reply gm4_ambrosia/data/gm4_ambrosia/functions/honey_bottle/give_new_bottle/level_1.mcfunction]: `/9` appears in all instances, though?

File: base/data/gm4-1.0/functions/load.mcfunction [hunk 3]
```diff
@@ -1,8 +1,8 @@
 execute unless data storage gm4:log queue run data merge storage gm4:log {queue:[]}
-data modify storage gm4:log queue append value {type:"text",message:'{"text":"[GM4]: Checking for updates...","color":"aqua"}'}
+data modify storage gm4:log queue append value {type:"text",message:'[{"translate":"%1$s%3427655$s","with":["[GM4]: ",{"translate":"text.gm4.prefix"}],"color":"aqua"},{"translate":"%1$s%3427655$s","with":["Checking for updates...",{"translate":"text.gm4.modules.update.check"}],"color":"aqua"}]'}
```
[the remark this answers is not in the document]
Comment (AndanteDevs) [reply base/data/gm4-1.0/functions/load.mcfunction]: Ok, I suppose that makes sense, will revert

File: gm4_ambrosia/data/gm4_ambrosia/functions/honey_bottle/give_new_bottle/level_1.mcfunction (hunk 2 above)
[the remark this answers is not in the document]
Comment (misode) [reply gm4_ambrosia/data/gm4_ambrosia/functions/honey_bottle/give_new_bottle/level_1.mcfunction]: But it isn't related to the translation. What happens when we want to rebalance the module and change that number to 6? Or have an upgrade that allows both 9 and 18?

[the remark this answers is not in the document]
Comment (AndanteDevs) [reply gm4_ambrosia/data/gm4_ambrosia/functions/honey_bottle/give_new_bottle/level_1.mcfunction]: Ah, right 👍 Understandable, will revert

[the remark this answers is not in the document]
Comment (Kroppeb) [reply gm4_ambrosia/data/gm4_ambrosia/functions/honey_bottle/give_new_bottle/level_1.mcfunction]: The "1" should be passed to the translation, incase a language wants to swap "word" order

[the remark this answers is not in the document]
Comment (misode) [reply gm4_ambrosia/data/gm4_ambrosia/functions/honey_bottle/give_new_bottle/level_1.mcfunction]: That's why I suggested `{"translate": "item.gm4.honey_bottle.lore", "with": ["1/9"]}`. If you want to go crazy, you can use `{"translate": "item.gm4.honey_bottle.lore", "with": ["1", "9"]}` and a translation of `%s/%s Spoons`.

[the remark this answers is not in the document]
Comment (AndanteDevs) [reply gm4_ambrosia/data/gm4_ambrosia/functions/honey_bottle/give_new_bottle/level_1.mcfunction]: If you want to do that and create a pull to my pull then please do, but my brain can't take all of these brackets :p

**Comment by Kroppeb** [comment]
> I was under the impression that `["",` was required for json text that was composed of multiple json objects. Is this not correct? I always remember needing it for the text to work.

I remember it being because for whatever reason all the other tags would copy the style of the first tag as if they were children of that tag. So everyone just placed "" as first element to fix that, even if it wasn't needed in that case. I don't know if that has been fixed or if it's just not noticeable.

**Comment by AndanteDevs** [comment]
In 99% of cases it's unnoticeable, yeah

**Comment by misode** [comment]
This PR seems good. I thing that I've mentioned this before, I don't think the `pack.mcmeta` files should use a translation for the description. It's technically still a placeholder so pointless, and since 1.15 we're no longer translating text that is only visible to admins.

**Comment by AndanteDevs** [comment]
All done! And there weren't any errors (@SpecialBuilder32)

**Pull request merged**